src/OrchardCore/OrchardCore.Localization.Core/LocalizationManager.cs

```diff
@@ -3,64 +3,64 @@
 using System.Globalization;
 using System.Linq;
 using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Extensions.Caching.Memory;
 
-namespace OrchardCore.Localization
+namespace OrchardCore.Localization;
+
+/// <summary>
+/// Represents a manager that manage the localization resources.
+/// </summary>
+public class LocalizationManager : ILocalizationManager
 {
+    private const string CacheKeyPrefix = "CultureDictionary-";
+
+    private static readonly PluralizationRuleDelegate _defaultPluralRule = n => (n != 1 ? 1 : 0);
+
+    private readonly IList<IPluralRuleProvider> _pluralRuleProviders;
+    private readonly IEnumerable<ITranslationProvider> _translationProviders;
+    private readonly IMemoryCache _cache;
+
     /// <summary>
-    /// Represents a manager that manage the localization resources.
+    /// Creates a new instance of <see cref="LocalizationManager"/>.
     /// </summary>
-    public class LocalizationManager : ILocalizationManager
+    /// <param name="pluralRuleProviders">A list of <see cref="IPluralRuleProvider"/>s.</param>
+    /// <param name="translationProviders">The list of available <see cref="ITranslationProvider"/>.</param>
+    /// <param name="cache">The <see cref="IMemoryCache"/>.</param>
+    public LocalizationManager(
+        IEnumerable<IPluralRuleProvider> pluralRuleProviders,
+        IEnumerable<ITranslationProvider> translationProviders,
+        IMemoryCache cache)
     {
-        private const string CacheKeyPrefix = "CultureDictionary-";
-
-        private static readonly PluralizationRuleDelegate _defaultPluralRule = n => (n != 1 ? 1 : 0);
-
-        private readonly IList<IPluralRuleProvider> _pluralRuleProviders;
-        private readonly IEnumerable<ITranslationProvider> _translationProviders;
-        private readonly IMemoryCache _cache;
+        _pluralRuleProviders = pluralRuleProviders.OrderBy(o => o.Order).ToArray();
+        _translationProviders = translationProviders;
+        _cache = cache;
+    }
 
-        /// <summary>
-        /// Creates a new instance of <see cref="LocalizationManager"/>.
-        /// </summary>
-        /// <param name="pluralRuleProviders">A list of <see cref="IPluralRuleProvider"/>s.</param>
-        /// <param name="translationProviders">The list of available <see cref="ITranslationProvider"/>.</param>
-        /// <param name="cache">The <see cref="IMemoryCache"/>.</param>
-        public LocalizationManager(
-            IEnumerable<IPluralRuleProvider> pluralRuleProviders,
-            IEnumerable<ITranslationProvider> translationProviders,
-            IMemoryCache cache)
+    /// <inheritdoc />
+    public async Task<CultureDictionary> GetDictionaryAsync(CultureInfo culture)
+    {
+        var cachedDictionary = _cache.GetOrCreate(CacheKeyPrefix + culture.Name, k => new Lazy<Task<CultureDictionary>>(async () =>
```

Comment:
This now caches not the dictionary but a `Task` that produces the dictionary. While this looks working at a first glance, this is not what should really happen. Rather, the dictionary itself needs to be cached. I don't think the `Lazy` makes sense here anyway.